### PR TITLE
CLI Layout and Create RayCluster function

### LINKED ISSUE #218: CLI Create Ray Cluster
CLI command allowing user to create a Ray Cluster specifying any parameters

### PR DESCRIPTION
**Pull request by carsonmh**
closes #218  

Creates layout, configuration, build configuration, and create raycluster command for the CLI. After building, get started with `codeflare --help` or just `codeflare`

### How to test
- Running unit tests `pytest -v tests/unit_test.py` or `python -m pytest tests/unit_test.py`
- After building SDK, you can run `codeflare create raycluster [options]` command. This should create the yaml file with the specified options. Options should be in the format: `--name=cluster-name` or `--machine_types='["gpu.large"]'`

### CODE REVIEW

File: tests/unit_test.py [hunk 1]
```diff
@@ -75,6 +77,37 @@
...
+# CLI testing
+def test_cli_working():
+    runner = CliRunner()
+    result = runner.invoke(cli)
+    assert result.exit_code == 0
+
+
+def test_cluster_creation_cli():
+    runner = CliRunner()
+    create_cluster_command = """
```
Comment (MichaelClifford) [inline tests/unit_test.py:89]: should this be renames to `define_cluster_command`?

**Review by MichaelClifford — CHANGES_REQUESTED** [review]
Really minor change otherwise looks good. Tested the CLI and define function as well. All seems to work as expected

:+1:

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
Off to a good start! Just one thing I noticed initially:
 - Running `codeflare create raycluster` for me causes `FileNotFoundError: [Errno 2] No such file or directory: '/Users/meyceoz/Documents/codeflare-sdk/src/codeflare_sdk/cli/commands/create.py'`. I realized after that the command was changed to `define` but the CLI should be able to handle erroneous input. For example, if i write `codeflare pizza`, right now it will tell me `FileNotFoundError: [Errno 2] No such file or directory: '/Users/meyceoz/Documents/codeflare-sdk/src/codeflare_sdk/cli/commands/pizza.py'`. Basically, just make sure you are checking that commands are valid too, not just options, and if not just tell the user it is invalid and to check `--help` for more details. I think `click` may have something for this as well.

**Comment by MichaelClifford** [comment]
@carsonmh looks good. Thanks for the changes. 

However, your commit history looks like you merged previous commits instead of re-baseing. Can you update this PR so that it only includes your commits? 

Thanks

**Comment by Maxusmusti** [comment]
@MichaelClifford it should be fixed now

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
LGTM, only one unit-test suggestion, then good to merge

**Review by Maxusmusti — APPROVED** [review]
LGTM

**Comment by Maxusmusti** [comment]
@MichaelClifford if it looks good to you, feel free to merge

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**